src/main/java/org/example/StudRepo.java

```diff
@@ -0,0 +1,170 @@
+package org.example;
+import java.time.LocalDate;
+import java.util.*;
+import java.time.*;
+/**
+ * Creating a Student repository with methods that validate inputs and throw exceptions if necessary
+ */
+public class StudRepo {
+    static LocalDate currentDate = LocalDate.now();
+    List<Student> students = new ArrayList<>();
+
+    public static void main(String[] args) {
+        StudRepo studRepo = new StudRepo();
+        try {
+            studRepo.addStudent("Andrea", "Hegyesi", LocalDate.of(1995, 7, 12), "F", "2950712313232");
+            studRepo.addStudent("John", "Snow", LocalDate.of(1988, 12, 24), "male", "1881224195796");
+            studRepo.addStudent("Arya", "Stark", LocalDate.of(2002, 2, 5), "FEMALE", "1920216418111");
+            studRepo.listStudentsByLastName();
+            studRepo.listStudentsByBirthDate();
+            studRepo.deleteStudent("1881224195796");
+            studRepo.listStudentsByLastName();
+            studRepo.retrieveByAge(27);
+        } catch (ValidationException e) {
+            System.out.println("Please input a valid student. " + e.getMessage());
+        }
+    }
+
+    /**
+     * Adding a student
+     *
+     * @param firstName    should not be empty
+     * @param lastName     should not be empty
+     * @param dateOfBirth  between 1900 and current year -18
+     * @param genderString should be male or female (or M and F), upper / lower case should both be accepted
+     * @param id           must have 13 characters, it is validated based on the last number
+     * @return list of students
+     * @throws ValidationException
+     */
+    public List<Student> addStudent(String firstName, String lastName, LocalDate dateOfBirth, String genderString, String id) throws ValidationException {
+        if ((firstName == null) || (firstName.equalsIgnoreCase("")) || (lastName == null)
+                || (lastName.equalsIgnoreCase(""))) {
```

Comment:
if the validation logic is large, create separate methods for it eg: valiadateName(student.getName)